sklearn/cluster/_agglomerative.py

```diff
@@ -169,6 +175,14 @@ def _single_linkage_tree(
 # Hierarchical tree building functions
 
 
+@validate_params(
+    {
+        "X": ["array-like"],
+        "connectivity": ["sparse matrix", None],
```

Comment:
The failing tests reveal that an array-like is allowed. Can you please revert and update the docstring ?

Reply:
Apparently, I did not run all tests when I propose the changes ;)
Changing the doc would then make the trick.

Reply:
Thanks guys:
- reverted removal of 'array-like'  commit in [78c097e](https://github.com/scikit-learn/scikit-learn/pull/25388/commits/78c097e9c11f2a8be3f0bfae8018e6ace905b848)
- updated docstring in [6900136](https://github.com/scikit-learn/scikit-learn/pull/25388/commits/69001365a4192dfe85703b3e1dbc40289c2a72f4)